README.rst

```diff
@@ -460,6 +460,161 @@ template ``context_processors`` in ``settings.py`` as shown below.
         },
     ]
 
+``register_menu_groups``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+It allow us to register different menu items or groups at different positions
+in the ``Main Navigation Menu``.
+
+``register_menu_groups`` takes a parameter of type (``dict``) in which key respresents the
+position of the menu item and its value needs to be a type of "dict".
+
+Code example:
+
+.. code-block:: python
+
+    from openwisp_utils.admin_theme.menu import register_menu_groups
+
+    my_group_items = {
+        1: {
+            'label': 'Add user',
+            'model': 'auth.User',
+            'name': 'changelist',
+            'icon': 'add-icon',
+        },
+        2: {'label': 'Code', 'url': 'https://openwisp.org/thecode.html'},
+    }
+
+    my_menu = {
+        1: {'label': 'My Group', 'items': my_group_items},
+        2: {
+            'model': 'test_project.Shelf',
+            'name': 'changelist',
+            'label': 'View Shelf',
+            'icon': 'shelf-icon',
+        },
+        3: {'label': 'Link Name', 'url': 'https://link.com'},
+    }
+
+    register_menu_groups(my_menu)
```

Comment:
I would have expected something like we have for the dashboard:
```
register_menu_group(position=1, config=group_config)
``` 

That way we can decide also the position of groups, which is not clear from this syntax.

Did you think about this?

Reply:
It can be a good option, then user will have call this function again and again. 
@nemesisdesign Would you like to go with this option? 
Also we will have to add ``menu group`` items like we are adding right now. I think current implementation is not having issue. User can understand by reading documentation. But your decision will be final.

Reply:
@niteshsinha17 let's define who's the user. For me users are users using OpenWISP.
This is meant for us, developers, users of this library as developers.
We will not be able to call this function only once, it will have to be called in each app, just like we do now with `register_menu_items`.

How would your solution work in practice?

Reply:
Yes, you are right. I was thinking about ``model link`` or ``general link``. For this, they will have to call functions multiple times.
Focusing on OpenWISP requirements your solution is good and it is consistent too. So let's stay with your soluton. I will make changes soon.